changelog/unreleased/kong/refactor_dns_client.yml

```diff
@@ -0,0 +1,3 @@
+message: refactor and implement a new DNS client library
```

Comment:
Could we add more information for this PR? such as the advantage or new feature of this dns client library.

Reply:
sure, will add more details later